src/Simulation/QSharpFoundation/Convert/Convert.qs

```diff
@@ -63,29 +63,42 @@ namespace Microsoft.Quantum.Convert {
     /// Converts a given big integer to an equivalent integer, if possible.
     /// The function returns a pair of the resulting integer and a Boolean flag
     /// which is true, if and only if the conversion was possible.
-    /// # Remarks
-    /// See [C# BigInteger constructor](https://docs.microsoft.com/dotnet/api/system.numerics.biginteger.-ctor?view=netframework-4.7.2#System_Numerics_BigInteger__ctor_System_Int64_) for more details.
     function MaybeBigIntAsInt(a : BigInt) : (Int, Bool) {
-        body intrinsic;
+        let arr = BigIntAsBoolArray(a);
```

Comment:
Would it be worth to do some range check before computing a possibly very large array just to determine not to perform the conversion?

Reply:
Good idea; we know the limits values, so it should be cheaper to calculate and compare the BigInts instead of construct the array.